relay-server/src/actors/envelopes.rs

```diff
@@ -1522,6 +1522,18 @@ impl EnvelopeProcessor {
             event._metrics = Annotated::new(metrics);
         }
 
+        if let Some(sampling_context) = envelope.sampling_context() {
+            if let Some(client_sample_rate) = sampling_context.sample_rate {
+                if let Some(ref mut contexts) = event.contexts.value_mut() {
+                    if let Some(EventContext::Trace(ref mut trace_context)) =
+                        contexts.get_context_mut("trace")
+                    {
+                        trace_context.client_sample_rate = Annotated::from(client_sample_rate.0);
```

Comment:
Two things about this:

1. During sampling, let's ensure we update the envelope header with the new sample rate. This allows a Relay to detect if the previous pipeline (regardless of whether that was a Relay or SDK) already applied the correct sample rate. Effectively, this makes sampling idempotent. 
2. This here is merely for informational purposes, so it can & should be moved into store normalization.

If we need the actual _client_ sample rate in the event, this means we need two headers for this.